### ISSUE TITLE
Expand template variables to include deep-copy references from other notes

### ISSUE THREAD

**Issue by al7arthi** [dendronhq/dendron#2263]
## Please select if your request is either something new or an enhancement ##

- [x] Enhancement of an existing Feature.
- [ ] Request of a new feature.

## Please select the area your request applies to. (Multiple selections are Possible. You can leave blank if you're not sure.) ##

- [ ] Workspace - VSCode workspace, vaults, Intellisense/autocomplete, Dendron settings
- [ ] Lookup - Dendron's Lookup Command
- [ ] Views - Dendron Preview, Tree View, Side Panels in the UI
- [ ] Schema - Dendron Schemas
- [ ] Pod - Data import from / export to Dendron
- [ ] Publish - External Site Publish
- [x] Markdown - Markdown features, such as syntax support and features

## Is your feature request related to a problem? Please describe ##

Not a problem but could be a very useful feature that supplements issue #2241 

## Describe the solution you'd like ##

I suggest expanding the template variables to include syntax for copying reference blocks from other notes. With deep copying instead of referencing, I can update or change the source content without affecting my previous notes and journals. This would allow including specific references in daily journal from other notes like a list of weekly affirmations that I can update weekly. In addition, I also suggest adding the ability to include date variables to the block reference, which can be useful in selecting certain items for certain days of the week as an example. Finally, if a line can be selected randomly from a block, that can be used to add a random quote of the day to daily journals.

## Describe alternatives you've considered ##

Hooks.

## Additional context ##

n.a

**Comment by hikchoi** [comment]
Thank you for the feature request.

Could you elaborate on what you mean by deep copy?

Do you mean grabbing the current state of the referenced source and resolving them into actual text where the template variables exist?

**Comment by hikchoi** [comment]
pinging @tma66 for visibility on template variable substitution enhancements.

**Comment by al7arthi** [comment]
> Thank you for the feature request.
> 
> Could you elaborate on what you mean by deep copy?
> 
> Do you mean grabbing the current state of the referenced source and resolving them into actual text where the template variables exist?

Yes, that is exactly what I meant. By resolving the variable into actual text, journal content won't change when the content of the referenced note is changed.
